features/steps/steps.go

```diff
@@ -265,9 +267,32 @@ func (c *Component) privateEndpointsAreNotEnabled() error {
 	return nil
 }
 
-func (c *Component) theDocumentInTheDatabaseShouldBe(id string, doc *godog.DocString) error {
-	// TODO: implement step for verifying documents stored in Mongo. No prior
-	// art of this being done properly in ONS yet so save to be done in future ticket
+// theDocumentInTheDatabaseShouldMatch checks if the filter in the DB matches the one provided.
+// Timestamps and the ETag are ignored.
+func (c *Component) theDocumentInTheDatabaseShouldMatch(fID string, doc *godog.DocString) error {
+	var expFilter model.Filter
+	if err := json.Unmarshal([]byte(doc.Content), &expFilter); err != nil {
+		return errors.Wrap(err, "failed to unmarshall")
+	}
+
+	ctx := context.Background()
+	col := c.cfg.FiltersCollection
+
+	var response model.Filter
+	if err := c.store.Conn().Collection(col).FindOne(ctx, bson.M{"filter_id": fID}, &response); err != nil {
+		return errors.Wrap(err, "failed to retrieve filter")
+	}
+
+	// ETags are compared by a separate assertion
+	response.ETag = ""
+	// Don't compare time, since it will be non-deterministic
+	response.LastUpdated = time.Time{}
```

Comment:
These fields should be predetermined in the tests as we're using the mock generator? (see features/mock/generator.go)

Reply:
Updated to use your revised version of the step method.